### ISSUE TITLE
Savon Model does not reset client attributes between instances

### ISSUE THREAD

**Issue by oranges13** [savonrb/savon#789]
I am using Savon to communicate with a SOAP endpoint that requires both standard and custom namespaces. When I initialize an instance of my Model class with certain data, until I close out the process and re-initialize the class the client settings remain the same -- even between different instances.

This is a major hassle and reduces the utility of using Savon Models by quite a bit.

For example:

```ruby
class MyClass
        extend Savon::Model
        def initialize(ns = nil)
		@altns = ns
		self.class.client wsdl: "http://www.wsdl.com/placeholder",
		convert_request_keys_to: :camelcase,
		env_namespace: :soap,
		strip_namespaces: false,
		namespaces: {"xmlns:wsdl" => (ns.nil? ? "http://schemas.microsoft.com/dynamics/2008/01/services" : ns)},
		headers: {"Content-Type" => "text/xml"},
		raise_errors: false,
		open_timeout: 240,
		read_timeout: 240
	end

	def find(params = {})
		super(
			message_tag: "#{params[:service_name]}ServiceFindRequest".to_sym,
			soap_action: (@altns.nil? ? "http://schemas.microsoft.com/dynamics/2008/01/services/#{params[:service_name]}Service/find" : "http://tempuri.org/#{params[:service_name]}Service/find"),
			message: {
				:query_criteria => {
					:@xmlns => "http://schemas.microsoft.com/dynamics/2006/02/documents/QueryCriteria",
					:criteria_element => params[:criteria]
				}
			}
		)
	end
end
```

So the big offense here is that if I do something like this:

```ruby
one = MyClass.new() # No extra namespace, so should use default
two = MyClass.new("http://tempuri.org") # Extra namespace!

one.find({:service_name => "SalesOrder", :criteria => [])
two.find({:service_name => "SalesOrder", :criteria => [])
```

The namespaces in the request are the same... even though they should be initialized with different values!

Savon generates *The same XML* for both requests, when the **xmlns:wsdl** namespaces should be different:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<wsdl:SalesOrderServiceFindRequest xmlns:wsdl="http://schemas.microsoft.com/dynamics/2008/01/services">
  <QueryCriteria xmlns="http://schemas.microsoft.com/dynamics/2006/02/documents/QueryCriteria"></QueryCriteria>
</wsdl:SalesOrderServiceFindRequest>
```

It's quite possible that I have written my `initialize` method incorrectly, and if that is the case please let me know, but otherwise this is a major issue that is causing me a lot of headache because I have to perform every request in a new process and cannot combine many requests in a single rake task for instance.

**Comment by oranges13** [comment]
Ok, I have actually figured this out in the mean time -- but the values within `initialize` remain the same across instances, which seems wrong.

If anyone else is needing to change namespaces or other attributes within the message between instances of Savon::Model, you can use the `attributes` method within your call to `super()` like so:

```ruby
def find(params = {})
		super(
			message_tag: "#{params[:service_name]}ServiceFindRequest".to_sym,
			attributes: {"xmlns:wsdl" => (@altns.nil? ? "http://schemas.microsoft.com/dynamics/2008/01/services" : @altns)},
			soap_action: (@altns.nil? ? "http://schemas.microsoft.com/dynamics/2008/01/services/#{params[:service_name]}Service/find" : "http://tempuri.org/#{params[:service_name]}Service/find"),
			message: {
				:query_criteria => {
					:@xmlns => "http://schemas.microsoft.com/dynamics/2006/02/documents/QueryCriteria",
					:criteria_element => params[:criteria]
				}
			}
		)
	end
```

**Comment by mikevoets** [comment]
This still needs to be fixed though!

**Comment by oranges13** [comment]
Just had to address this issue *again* in a project. I added the `@client = nil` in my model but that still doesnt' appear to help.

As you can see in my previous post, I was setting the attributes property to get the namespace correct between requests. Some of my methods (find, read) had this set explicitly, while others did not because they could safely default. However, I just discovered that between these types of requests, the attributes (and thus the namespace) was being carried over between requests, even though I'm instantiating a new model for each one!

Long story short, this is still an issue. I had to manually hard-code the attributes property for all my request endpoints, not just the ones I needed to customize, otherwise the custom attributes might bleed over into subsequent requests.

**Comment by oranges13** [comment]
This is still an issue and needs to be addressed by the developers.
